CMakeLists.txt

```
@@ -77,10 +77,17 @@ add_library(${PROJECT_NAME} INTERFACE)
 #  INTERFACE_COMPILE_FEATURES cxx_std_11
 #)
 
+option(USE_SYSTEM_WISE_TL_OPTIONAL "Use a system wise installation of tl-optional" OFF)
+
+if(USE_SYSTEM_WISE_TL_OPTIONAL)
+  find_package(tl-optional REQUIRED)
+endif()
```

A system installation of `tl-optional` is *only* used when this option is set. I'd rather prefer that it is used whenever available. Something along the lines of,
```suggestion
find_package(tl-optional 1.0.0 QUIET)
if(USE_SYSTEM_WISE_TL_OPTIONAL AND NOT tl-optional_FOUND)
  message(FATAL_ERROR "Could not find required package tl-optional")
endif()
```